qiskit/chemistry/algorithms/pes_samplers/bopes_sampler.py

```diff
@@ -113,7 +116,8 @@ def sample(self, driver: BaseDriver, points: List[float]) -> BOPESSamplerResult:
         self._energies = []
         for key in self._raw_results:
             energy = self._raw_results[key].computed_energies[0] + \
-                     self._raw_results[key].nuclear_repulsion_energy
+                     self._raw_results[key].nuclear_repulsion_energy + \
+                     (self._raw_results[key].frozen_extracted_energy if self._frozen_energy else 0)
```

Comment:
Is there a reason to not want to include any frozen core energy. It seems the same 'problem' with frozen core would exist if particle hole transform is used as that extracts energy that needs to be replaced for the total ground state. Am not sure why this here does not just use the energy from the result that sums up whatever parts were extracted etc. @pbark

Reply:
@pbark Tagging again as a reminder of the above comment

Reply:
@woodsp-ibm sorry for the late response. You are right that the same problem would exist also in the energy shift from the particle hole transformations. I will push a fix to this here later this week. Thanks for pointing this! The rest looks good to me! Thanks @molar-volume !

Reply:
May I suggest the following (I cannot propose a suggestion because one line above is unreachable by this feature of Github):
```
energy = self._raw_results[key].total_energies[0]
```

Note that `total_energies = electronic_energies + nuclear_repulsion_energy` and `electronic_energies = compute_energies + frozen_extracted_energy + ph_extracted_energy`. Thus, all edge cases should be covered by this.